providers/implementations/ciphers/cipher_aes_siv_hw.c

```diff
@@ -25,6 +25,9 @@ static int aes_siv_initkey(void *vctx, const unsigned char *key, size_t keylen)
     OPENSSL_CTX *libctx = ctx->libctx;
     const char *propq = NULL;
 
+    EVP_CIPHER_free(ctx->cbc);
+    EVP_CIPHER_free(ctx->ctr);
+
```

Comment:
I know it's probably a "can't happen" case but with the default case below you keep dangling pointers in ctx->cbc and ctx->ctr, they should be NULLified.

Reply:
fair enough - I also changed the default so it just falls through now.